plugins/http_extract/src/http_extract/component.py

```
@@ -0,0 +1,580 @@
+        if self._session is None:
+            raise ValueError()
+
+        for _attempt in range(self.config.retries + 1):
+            try:
+                async with self._session.request(
+                    self.config.method, url, params=params, headers=headers
+                ) as response:
+                    if response.status not in self.config.valid_status_codes:
+                        if self.config.ignore_http_errors:
+                            logger.warning(
+                                f"HTTP {response.status} ignored: {url}"
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)